engine/rpcserver.go

```diff
@@ -1832,39 +1889,290 @@ func (s *RPCServer) GCTScriptListAll(context.Context, *gctrpc.GCTScriptListAllRe
 	if !gctscript.GCTScriptConfig.Enabled {
-		return &gctrpc.GCTScriptGenericResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+		return &gctrpc.GenericResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
 	}
 
 	err := gctscript.ShutdownAll()
 	if err != nil {
-		return &gctrpc.GCTScriptGenericResponse{Status: "error", Data: err.Error()}, nil
+		return &gctrpc.GenericResponse{Status: "error", Data: err.Error()}, nil
 	}
 
-	return &gctrpc.GCTScriptGenericResponse{
+	return &gctrpc.GenericResponse{
 		Status: MsgStatusOK,
 		Data:   "all running scripts have been stopped",
 	}, nil
 }
 
 // GCTScriptAutoLoadToggle adds or removes an entry to the autoload list
-func (s *RPCServer) GCTScriptAutoLoadToggle(ctx context.Context, r *gctrpc.GCTScriptAutoLoadRequest) (*gctrpc.GCTScriptGenericResponse, error) {
+func (s *RPCServer) GCTScriptAutoLoadToggle(_ context.Context, r *gctrpc.GCTScriptAutoLoadRequest) (*gctrpc.GenericResponse, error) {
 	if !gctscript.GCTScriptConfig.Enabled {
-		return &gctrpc.GCTScriptGenericResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+		return &gctrpc.GenericResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
 	}
 
 	if r.Status {
 		err := gctscript.Autoload(r.Script, true)
 		if err != nil {
-			return &gctrpc.GCTScriptGenericResponse{Status: "error", Data: err.Error()}, nil
+			return &gctrpc.GenericResponse{Status: "error", Data: err.Error()}, nil
 		}
-		return &gctrpc.GCTScriptGenericResponse{Status: "success", Data: "script " + r.Script + " removed from autoload list"}, nil
+		return &gctrpc.GenericResponse{Status: "success", Data: "script " + r.Script + " removed from autoload list"}, nil
 	}
 
 	err := gctscript.Autoload(r.Script, false)
 	if err != nil {
-		return &gctrpc.GCTScriptGenericResponse{Status: "error", Data: err.Error()}, nil
+		return &gctrpc.GenericResponse{Status: "error", Data: err.Error()}, nil
+	}
+	return &gctrpc.GenericResponse{Status: "success", Data: "script " + r.Script + " added to autoload list"}, nil
+}
+
+// SetExchangeAsset enables or disables an exchanges asset type
+func (s *RPCServer) SetExchangeAsset(_ context.Context, r *gctrpc.SetExchangeAssetRequest) (*gctrpc.GenericSubsystemResponse, error) {
+	exch := GetExchangeByName(r.Exchange)
+	if exch == nil {
+		return nil, errExchangeNotLoaded
+	}
+
+	exchCfg, err := Bot.Config.GetExchangeConfig(r.Exchange)
+	if err != nil {
+		return nil, err
+	}
+
+	base := exch.GetBase()
+	if base == nil {
+		return nil, errExchangeBaseNotFound
+	}
+
+	if r.Asset == "" {
+		return nil, errors.New("asset type must be specified")
+	}
+
+	a := asset.Item(r.Asset)
+	err = base.CurrencyPairs.SetAssetEnabled(a, r.Enable)
+	if err != nil {
+		return nil, err
+	}
+	err = exchCfg.CurrencyPairs.SetAssetEnabled(a, r.Enable)
+	if err != nil {
+		return nil, err
+	}
+
+	return &gctrpc.GenericSubsystemResponse{}, nil
```

Comment:
Can we please add a response here even if its "true"?
Just to let the user know it was successful